content/zh/docs/v3.0/concepts/extensibility.md

```diff
@@ -4,4 +4,59 @@ title: "如何扩展 Dubbo"
 linkTitle: "扩展性"
 weight: 6
 description: "Dubbo 通过 SPI 机制提供了非常灵活的可扩展性"
----
\ No newline at end of file
+---
+
+## 扩展设计理念
+
+可扩展性是任何一个系统所追求的，对于 Dubbo 来说是同样适用。
+
+### 微核心，插件式，平等对待第三方
+
+一般来说，系统会采用 Factory、IoC、OSGI 等方式管理插件生命周期。考虑到 Dubbo 的适用面，不想强依赖 Spring 等 IoC 容器。
+而自己造一个小的 IoC 容器，也觉得有点过度设计，所以选择最简单的 Factory 方式管理插件。在 Dubbo 中，所有内部实现和第三方实现都是平等的。
+
+### 每个扩展点只封装一个变化因子，最大化复用
+
+每个扩展点的实现者，往往都只是关心一件事。如果用户有需求需要进行扩展，那么只需要对其关注的扩展点进行扩展就好，极大的减少用户的工作量。
```

Comment:
Maybe, this part should belong to the above chapter `What's the benifits of system extension?`

Reply:
done